packages/upload-api/src/blob/accept.js

```diff
@@ -0,0 +1,62 @@
+import * as Server from '@ucanto/server'
+import * as DID from '@ipld/dag-ucan/did'
+import * as W3sBlob from '@web3-storage/capabilities/web3.storage/blob'
+import { Assert } from '@web3-storage/content-claims/capability'
+import { create as createLink } from 'multiformats/link'
+import { Digest } from 'multiformats/hashes/digest'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { code as rawCode } from 'multiformats/codecs/raw'
+import * as API from '../types.js'
+import { AllocatedMemoryHadNotBeenWrittenTo } from './lib.js'
+
+const R2_REGION = 'auto'
+const R2_BUCKET = 'carpark-prod-0'
+
+/**
+ * @param {API.W3ServiceContext} context
+ * @returns {API.ServiceMethod<API.BlobAccept, API.BlobAcceptSuccess, API.BlobAcceptFailure>}
+ */
+export function blobAcceptProvider(context) {
+  return Server.provideAdvanced({
+    capability: W3sBlob.accept,
+    handler: async ({ capability }) => {
+      const { blob, space } = capability.nb
+      // If blob is not stored, we must fail
+      const hasBlob = await context.blobsStorage.has(blob.digest)
+      if (hasBlob.error) {
+        return hasBlob
+      } else if (!hasBlob.ok) {
+        return {
+          error: new AllocatedMemoryHadNotBeenWrittenTo(),
+        }
+      }
+
+      // TODO: we need to support multihash in claims, or specify hardcoded codec
+      const digest = new Digest(sha256.code, 32, blob.digest, blob.digest)
```

Comment:
This seems incorrect because because bytes and digest in the constructor aren't the same because `bytes` are supposed to be multihash prefixed byte array and and `digest` parameter supposed to be bytes without multihash prefix.

I think you really should just use `decode(blob.digest)` exported by the [digest module](https://github.com/multiformats/js-multiformats/blob/master/src/hashes/digest.ts#L21-L25) it will also automatically figure out hashing algorithm etc...